### DISCUSSION TITLE
Ejbca8 failed with error No Jakarta Enterprise Beans

### DISCUSSION THREAD

**Discussion by hanuamantha** [Keyfactor/ejbca-ce#607] [Q&A]
I am trying to setup ejbca8 with  WildFly Full 26.1.3.Final (WildFly Core 18.1.2.Final).
I am facing below errors.

```

2024-06-12 02:53:18,232 ERROR [org.jboss.as.controller.management-operation] (Controller Boot Thread) WFLYCTL0013: Operation ("deploy") failed - address: ([("deployment" => "ejbca.ear")]) - failure description: {
    "WFLYCTL0080: Failed services" => {"jboss.deployment.subunit.\"ejbca.ear\".\"ejbca-ejb.jar\".INSTALL" => "WFLYSRV0153: Failed to process phase INSTALL of subdeployment \"ejbca-ejb.jar\" of deployment \"ejbca.ear\"
    Caused by: org.jboss.as.server.deployment.DeploymentUnitProcessingException: WFLYEE0052: Failed to install component ClearCacheSessionBean
    Caused by: org.jboss.as.server.deployment.DeploymentUnitProcessingException: WFLYEJB0406: No Jakarta Enterprise Beans found with interface of type 'org.ejbca.core.protocol.msae.KecCache' for binding org.ejbca.core.ejb.config.ClearCacheSessionBean/kecCache"},
    "WFLYCTL0412: Required services that are not installed:" => [
        "jboss.deployment.subunit.\"ejbca.ear\".\"ejbca-ejb.jar\".component.EndEntityManagementSessionBean.VIEW.\"org.ejbca.core.ejb.ra.EndEntityManagementSessionLocal\".LOCAL",
        "jboss.naming.context.java.comp.ejbca.ejbca-ejb.RaMasterApiSessionBean",
        "jboss.naming.context.java.comp.ejbca.ejbca-ejb.OcspKeyRenewalSessionBean",
        "jboss.naming.context.java.comp.ejbca.ejbca-ejb.KeyStoreCreateSessionBean",
        "jboss.naming.context.java.comp.ejbca.ejbca-ejb.EjbcaWSHelperSessionBean",
        "jboss.deployment.subunit.\"ejbca.ear\".\"ejbca-ejb.jar\".component.ServiceDataSessionBean.VIEW.\"org.ejbca.core.ejb.services.ServiceDataSessionLocal\".LOCAL",
        "jboss.naming.context.java.comp.ejbca.ejbca-ejb.ImportCrlSessionBean",
        "jboss.deployment.subunit.\"ejbca.ear\".\"ejbca-ejb.jar\".component.PublishingCrlSessionBean.VIEW.\"org.ejbca.core.ejb.crl.PublishingCrlSessionLocal\".LOCAL",
        "jboss.naming.context.java.comp.ejbca.ejbca-ejb.AuthorizationSystemSessionBean",
        "jboss.naming.context.java.comp.ejbca.ejbca-ejb.LegacyRoleManagementSessionBean",
        "jboss.deployment.subunit.\"ejbca.ear\".\"ejbca-ejb.jar\".component.ImportCrlSessionBean.VIEW.\"org.ejbca.core.ejb.crl.ImportCrlSessionLocal\".LOCAL",
        "jboss.naming.context.java.comp.ejbca.ejbca-ejb.PublisherQueueSessionBean",
        "jboss.deployment.subunit.\"ejbca.ear\".\"ejbca-ejb.jar\".component.SignSessionBean.VIEW.\"org.ejbca.core.ejb.ca.sign.SignSessionLocal\".LOCAL",
        "jboss.naming.context.java.comp.ejbca.ejbca-ejb.OcspResponseGeneratorSessionBean",
        "jboss.naming.context.java.comp.ejbca.ejbca-ejb.EjbcaRestHelperSessio
```nBean",

**Comment by primetomas** [comment]
There will be a new community release tomorrow (8.3). Please check back then and try the new release. This should be fixed there.

**Reply by hanuamantha** [reply]
okay

**Reply by hanuamantha** [reply]
Hi @primetomas , After using ejbca 8.3.1 I am seeing below error .
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
04:24:18,182 INFO  [org.apache.cxf.wsdl.service.factory.ReflectionServiceFactoryBean] (MSC service thread 1-1) Creating Service {http://ws.protocol.core.ejbca.org/}EjbcaWSService from class org.ejbca.core.protocol.ws.EjbcaWS
04:24:19,898 INFO  [org.apache.cxf.endpoint.ServerImpl] (MSC service thread 1-1) Setting the server's publish address to be http://192.168.0.10:9443/ejbca/ejbcaws/ejbcaws
ERROR StatusLogger Reconfiguration failed: No configuration found for '277050dc' at 'null' in 'null'
ERROR StatusLogger Reconfiguration failed: No configuration found for '277050dc' at 'null' in 'null'
ERROR StatusLogger Reconfiguration failed: No configuration found for '277050dc' at 'null' in 'null'
ERROR StatusLogger Reconfiguration failed: No configuration found for '277050dc' at 'null' in 'null'
ERROR StatusLogger Reconfiguration failed: No configuration found for '277050dc' at 'null' in 'null'

**Reply by primetomas** [reply]
That is strange, I haven't seen that by anyone else. It deploys for you now, but you see those log items in the server.log when running on WildFly 26. Is that correct?

**Reply by hanuamantha** [reply]
In container logs I can see the above error(i.e., `docker logs ejbca -f`) and even when I run ejbca help command.
`(ejbca)[root@ubuntu_vm /]# ejbca --help
ERROR StatusLogger Reconfiguration failed: No configuration found for '277050dc' at 'null' in 'null'
`

**Reply by primetomas** [reply]
Is that the EJBCA container from DockerHub? Or some container you created yourself?
There are some known bugs in WildFly that created log entries that are harmless.

**Reply by hanuamantha** [reply]
from source code of wildfly 26 and ejcba 8.3.1, built ejbca container.  Not from DcokerHub. You mean we can ignore the errors.

**Reply by primetomas** [reply]
Then you can always compare your configuration with the one in the container from Dockerhub.

**Reply by hanuamantha** [reply]
I have compared standalone.conf, both are same. Any other configuration files do I need to compare?

**Reply by primetomas** [reply]
Not that I am aware of. Perhaps you need to live with those log entries. How many of those is it?

**Reply by hanuamantha** [reply]
15 entries are present.

**Reply by primetomas** [reply]
Looks to be related to logging configuration:
https://stackoverflow.com/questions/66545546/error-statuslogger-reconfiguration-failed-no-configuration-found-for-73d16e93

**Reply by hanuamantha** [reply]
let me check this and get back to you

**Comment by hanuamantha** [comment]
After adjusting log file I am able to bring up ejbca. Thanks.
